### ISSUE TITLE
I'm having trouble with NetGSA

### ISSUE THREAD

**Issue by Ju-Rien** [lgeistlinger/EnrichmentBrowser#4]
Hi again,

This time I am reviewing `.netgsa()` code to understand why I get error messages.

### Line 573
When performing NetGSA, with my personal data sets and with a `makeExampleData()` data set, I encounter this message:
```
Error in edgelist2adj(out.file, vertex.names = unique(as.vector(grn[,  : 
  The adjacency matrix is not lower triangular! Check ordering of the variables!
```
which pops at this line in the code:
https://github.com/lgeistlinger/EnrichmentBrowser/blob/8d632efeaf29d20bc50e4c676951de0d7b776d87/R/nbea.R#L573

With the generated data set, I call `netgsa()` with:
```
set.seed(42)
res.netgsa <- nbea(method = "netgsa", se = ex.se, gs = ex.gs, grn = ex.grn,
                   perm = 1000, alpha = 0.1)
```
with the data being generated by:
```
set.seed(42)
ex.se <- makeExampleData(what = "SE", type = "ma", nfeat = 1000,
                               nsmpl = 50, blk = FALSE, norm = FALSE, 
                               de.ana = TRUE)
set.seed(42)
ex.gs <- makeExampleData(what = "gs", gnames = rownames(ex.se), n = 60, 
                         min.size = 5, max.size = 200)

set.seed(42)
ex.grn <- makeExampleData(what = "grn", nodes = sample(rownames(ex.se), 200), 
                          edge.node.ratio = 2)
```

I tried ordering the genes in the GRN matrix, without effects. 

Now, interestingly, when I overwrite the netgsa function `edgelist2adj()` to make an **undirected** adjacency matrix, the error disappears (but I get another error further downstream). I overwrite the function with:
```
edgelist2adj <- function(file, vertex.names, mode = "undirected"){
        netgsa::edgelist2adj(file = file, vertex.names = vertex.names, mode = mode)
}
```
The argument `directed=TRUE` is given to the `NetGSA()` function call, so I believe that this shouldn't be the solution. I don't know what to suggest as a fix for this. 

### Line 578
Further downstream, I get an error message for subscript out of bounds at:

https://github.com/lgeistlinger/EnrichmentBrowser/blob/8d632efeaf29d20bc50e4c676951de0d7b776d87/R/nbea.R#L578

It seems to me that the attempted accession is on the wrong dimension. The object `cmat`, as created by these lines
https://github.com/lgeistlinger/EnrichmentBrowser/blob/e8f8d31dddb65cda934e79bb6cd63dc6a887f3d7/R/nbea.R#L562-L566
has rows names as gene names, and column names as gene sets names. See the following lines to quickly see my point.
```
> head(rownames(adjm))
[1] "g488" "g606" "g786" "g808" "g317" "g478"

> head(colnames(cmat))
[1] "gs1" "gs2" "gs3" "gs4" "gs5" "gs6"
> head(rownames(cmat))
[1] "g1"    "g10"   "g100"  "g1000" "g101"  "g102" 
```
I suggest that the line 578 should read:
`cmat <- cmat[rownames(adjm), ]`
instead of
`cmat <- cmat[, rownames(adjm)]`.

I currently use the NetGSA package version 3.0. .

Thank you for the fast feedback, it's really appreciated!

**Comment by lgeistlinger** [comment]
Hi,

Thanks for the thorough testing of the `EnrichmentBrowser`!

The `netgsa` package has apparently been reworked lately, including modifications of the `edgelist2adj` function as well as the apparent removal of the `covsel` function.

I have contacted the author of the `netgsa` package on these issues and keep you posted once I know more.

**Comment by ktmf1991** [comment]
Hi,

May I ask if there is any plan of rewriting the part for `netgsa`, just out of curiosity?

Many thanks!

**Comment by lgeistlinger** [comment]
Yes, I am about to move on from evaluating set-based enrichment methods

https://www.biorxiv.org/content/10.1101/674267v1 (manuscript currently at revision stage at Briefings in Bioinformatics)

to assessing the performance of network-based methods.

I can't give you an exact date, but I assume to start working on the `nbea` module including `netgsa` at some point in Nov (= next month).

**Comment by lgeistlinger** [comment]
Fixed via bd22f9e
Will become effective in EnrichmentBrowser v2.16.1 (release) and v2.17.2 (devel).
